### PR TITLE
Add coveralls coverage checker in CI

### PR DESCRIPTION
**Pull request by jamie-lynch**
## What is the purpose of this change?

This change adds coverage checking to the CI checks to ensure that future changes don't negatively effect test coverage.

## What does this change?

- Add `test:coverage` and `validate:ci` scripts to `package.json`
- Add [coverallsapp/github-action](https://github.com/coverallsapp/github-action) action to _ci_ workflow
- Update _ci_ workflow to run `validate:ci`
- Add `coverage/` to _.gitignore_

### CODE REVIEW

**Review by SiAdcock — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -36,8 +36,10 @@
     "storybook": "start-storybook -p 6006",
     "tag:beta:all": "ts-node ./scripts/tag-beta-all",
     "test": "jest",
+    "test:coverage": "jest --coverage",
     "tsc": "tsc",
     "validate": "yarn tsc && yarn lint && yarn test",
+    "validate:ci": "yarn tsc && yarn lint && yarn test:coverage --ci",
```
Comment (SiAdcock) [inline package.json:42]:
```suggestion
    "validate:ci": "yarn validate && yarn test:coverage --ci",
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,5 +16,9 @@ jobs:
       - name: build
         run: |
           yarn install --frozen-lockfile
-          yarn validate
+          yarn validate:ci
           yarn ci:build
```
Comment (SiAdcock) [inline .github/workflows/ci.yml:20]: Aside: we should probably rename this script to `build:ci`

**Review by SiAdcock — APPROVED** [review]
Blessings 🙏

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jamie-lynch) [reply .github/workflows/ci.yml:20]: Yeah, that'd be good. I'll fire another PR over with that in.

File: package.json (hunk 1 above)
Comment (jamie-lynch) [reply package.json:42]: This would mean running the tests twice which I'm not such a fan of, but then less duplication is good so 🤷‍♂️

**Pull request merged**